### ISSUE TITLE
Training search space: make batch_size tunable, widen initial_lr, derive the budget

### ISSUE THREAD

**Issue by matthiaskloft** [matthiaskloft/bayesflow-hpo#69]
## Summary

The training search space fixes two settings that a measured grid says should not be fixed, and cannot express the one budget that actually matters. Follow-on to #67, which established the schedule/stopping modes; this issue is the search space itself.

Evidence: a 3x3 batch x learning-rate grid in `bayesflow-irt` — 9 arms x 3 seeds, all successful, simulation-matched at 102,400 simulations on one architecture. Full results in https://github.com/matthiaskloft/bayesflow-hpo/issues/67#issuecomment-5349211054.

## 1. `batch_size` is a constant

```python
# search_spaces/training.py
batch_size: IntDimension = field(
    default_factory=lambda: IntDimension("batch_size", constant=256)
)
```

At a constant batch, the batch x learning-rate interaction is invisible by construction — and it was the largest effect measured. The penalty for reusing a learning rate tuned at a smaller batch grows with batch size:

| batch | `b` calibration at lr 5e-4 | at its best lr | ratio |
| ----- | -------------------------- | -------------- | ----- |
| 32 | 0.0602 | 0.0111 | 5.4x |
| 64 | 0.0765 | 0.0186 | 4.1x |
| 128 | 0.1063 | 0.0298 | 3.6x |

Batch size is also a throughput knob once the learning rate moves with it: at its own best lr, batch 128 cost 268 s/run against 490 s/run at batch 32 — 1.83x — for a recovery difference of 0.003, inside seed noise (sd ~0.005-0.009).

## 2. `initial_lr` is capped at 5e-3

```python
initial_lr: FloatDimension = field(
    default_factory=lambda: FloatDimension("initial_lr", low=1e-4, high=5e-3, log=True)
)
```

Three of the four Pareto-optimal cells in the grid used `8e-3`, so the cheap end of the front is outside the search space entirely. The front over (recovery up, calibration down, wall-clock down) was `bs32 lr2e-3`, `bs32 lr8e-3`, `bs64 lr8e-3`, `bs128 lr8e-3`.

`8e-3` is not a free win and the range should be widened rather than re-centred: it improves `b` calibration at every batch size, but is the worst of the three rates for `log(a)` calibration at the five-item corner (0.0287/0.0306/0.0391 vs ~0.019 at 5e-4).

Note also that `lr 5e-4` contributed nothing: all three of its arms are dominated, and `bs128 lr5e-4` is dominated on every axis at once — worst recovery, worst calibration, and no cheaper than `bs128 lr8e-3` at the same 266 s.

## 3. Couple lr to batch by reparametrisation, not a conditional space

```
lr = lr_ref * (batch_size / reference_batch)
```

The coupling has a known functional form — Smith et al. (2018), `B` proportional to `epsilon` — so a rectangle in decoupled coordinates is easier for TPE than a conditional space. TPE models parameters marginally, which is the worst case for a ridge. This is a restatement of the position already taken in #67 under "Explicitly not requested", now with the interaction measured rather than assumed.

## 4. Derive `num_batches` from a simulation budget

In online SBI `batch_size` is simultaneously the data-volume knob: `simulations = batch_size * steps`. At most two of {batch, steps, simulations} can be fixed. Fixing the **simulation budget** is the meaningful choice, since it is proportional to GPU-seconds:

```
num_batches = sims_budget // (batch_size * epochs)
```

This also removes the lr x epochs ridge described in #67: under cosine annealing the optimizer's total displacement scales roughly with `lr_peak * steps`, so tuning both is close to redundant. Measured: `lr=2e-3` at 16 epochs matched `lr=5e-4` at 32 epochs, and 64 epochs added nothing further.

## Explicitly not in scope: warmup length as a tunable dimension

Warmup is currently hardcoded at 5% of total steps and has never been swept. It becomes tempting to tune precisely when the lr range widens, since warmup is the mechanism that makes high learning rates survivable — so this needs stating rather than leaving implicit.

The argument for tuning it is Shallue et al. (2019, Sec. 5): "the optimal values of other metaparameters, such as the momentum and learning rate decay schedule, should not be assumed to remain constant or scale in a simple way as the batch size increases."

The argument against is stronger and comes from the same paper's limitations (Sec. 5.1):

> "We are most confident in our search spaces that tuned the fewest metaparameters (such as in our experiments that only tuned learning rate and momentum). We found it quite difficult to be confident that our tuning was sufficient when we searched over learning rate decay schedules... our search spaces may have been too wide at some batch sizes."

That is the authors of the definitive batch-size study reporting that adding schedule parameters is what made their own tuning unreliable, at a budget far larger than ours. Adding a third correlated axis to `{batch, lr}` is the change most likely to degrade the rest of the search.

**Treatment**: sweep warmup factorially *outside* the HPO and fix it at a known-good constant. A design that also settles a pending question — warmup fraction `{0, 0.02, 0.05, 0.15}` x lr `{2e-3, 8e-3}` at one batch size, on the architecture with `num_outer_sabs=2`, which is the architecture `8e-3` has not been tested on. 8 cells x 3 seeds, roughly 2 GPU-hours.

## Also worth carrying over from the grid

Recovery correlation alone is not a safe HPO objective. `bs128 lr5e-4` ranks 4th of 9 on recovery while being last of 9 on calibration in all eight measured cells. Whatever objective this space is searched against, a single point-estimate recovery metric would select that cell.

## Dependency

Should land after `fix/hpo-review-findings`, which already rewrites dimension types (`BoolDimension`) and removes dead dimensions in three search spaces. Stacking this on top beforehand would compound an unreviewed branch.

Tracked in `docs/TODO.md` as Package K.

— Claude Code (Opus 5)
